api/client/trust.go

```diff
@@ -0,0 +1,323 @@
+package client
+
+import (
+	"bufio"
+	"bytes"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"net"
+	"net/http"
+	"net/url"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/docker/distribution/digest"
+	"github.com/docker/docker/pkg/homedir"
+	flag "github.com/docker/docker/pkg/mflag"
+	"github.com/docker/docker/pkg/term"
+	"github.com/docker/docker/pkg/tlsconfig"
+	"github.com/docker/docker/registry"
+	"github.com/docker/notary/client"
+)
+
+var untrusted bool
+
+func addTrustedFlags(fs *flag.FlagSet) {
+	var trusted bool
+	if e := os.Getenv("DOCKER_NOTARY"); e != "" {
+		var err error
+		trusted, err = strconv.ParseBool(e)
+		if err != nil {
+			// treat any other value as true
+			trusted = true
+		}
+	}
+	fs.BoolVar(&untrusted, []string{"-untrusted"}, !trusted, "Use or update trusted repositories on push and pull")
+}
+
+func isTrusted(name string) bool {
+	// TODO(dmcgowan): whitelist specific names to be untrusted
+	// TODO(dmcgowan): canonicalize name before checking whitelist
+	return !untrusted
```

Comment:
@dmcgowan what would be examples of automatically untrusted names? should it be related to insecure registries (aka resolves to 127/8 or was configured)